doc/source/report/classification.rst

```diff
@@ -111,48 +134,109 @@ be used as test data to evaluate the model performance. Notice that we
+checked that whether our silhouette score is appropriate.
 
-1. Clean up the existing strain\_classification.py: create functions and
-   objects.
-2. Adding new models: knn, random forests, neural networks, logistic
-   regressions.
-3. Doing unsupervised learning: k-means.
-4. Compute confidence interval for the accuracy of each model.
-5. Give insights about how mice behavior is related to their genetic
-   heritage (strain difference).
+Result
+-------------
+**Classification**
+
+**Clustering**
+***K-means***
+The silhouette scores corresponding to the number of clusters ranging from 2 to 16 
+are:[0.835, 0.775, 0.423, 0.415, 0.432, 0.421, 0.404, 0.383, 0.421, 0.327, 0.388, 0.347, 0.388, 0.371,0.362]
+ . We plot 6 clusters here to show, and found that Czech and CAST mice behaved 
+ quite differently from each other. 
+.. plot:: report/plots/plot_km_result.py
+   Distribution of strains in clusters by K-means algorithm
+
+
+***Hierarchical clustering***
+
+The optimal distance measure is l1-norm and the optimal linkage method is average linkage method. The silhouette scores corresponding to the number of clusters ranging from 2 to 16 are:  0.8525, 0.7548, 0.7503, 0.6695, 0.6796, 0.4536, 0.4557, 0.4574, 0.3997, 0.4057, 0.3893, 0.3959, 0.4075, 0.4088, 0.4179. It seems 6 clusters is a good choice from the silhouette scores. 
+However, the clustering dendrogram tells a different story. Below shows the last 10 merges of the hierarchical clustering algorithm. The black dots indicate the earlier merges. The leaf texts are either the mouse id (ranges from 0 to 169) or the number of mice in that leaf. Clearly, we see that almost all the mice are clustered in 2 clusters, very far from the rest individuals. Thus, the hierarchical clustering fails to correctly cluster the mice in the case. 
+.. plot:: report/plots/plot_hc_dendrogram.py
+
+   Dendrogram of the hierarchical clustering
+The failure of the the algorithm might be due to the different importance levels of the features in determining which cluster a mouse belongs to. One improvement could be that using only the important features determined in the classification algorithms to cluster the mice, but given the unsupervised learning nature of the algorithm, not using the results from the classification is fair for clustering tasks.
+The distribution of strains in each cluster in the case of using 6 clusters are shown below. Obviously, the mice almost fall into the same cluster.
+.. plot:: report/plots/plot_hc_result.py
+
+   Distribution of strains in clusters by agglomerative hierarchical clustering
+
+Future work
+----------------
+The future research should focus more on feature engineering, including the questions 
+that whether more features could be added to the model. Moreover, an economized subset 
+of features should be evaluated to see whether we can reduce the model complexity
+without losing too much model accuracy. 
```

Comment:
Yes, I have just noticed that we are planning to use the economized set of features. I will probably change it depending on the progress of the classification part.